schedule/schedule-appengine/src/main/client/jasify/admin/activity/admin-activity.js

```diff
@@ -123,8 +136,20 @@
 
         function create() {
             Activity.add(vm.activity).then(ok, fail);
+
             function ok(r) {
                 vm.alert('info', 'Activity created!');
+                // TODO This should be server side work
+                if (vm.repeatType == "Weekly") {
+                    while (vm.repeatDate && vm.activity.start.getTime() < vm.repeatDate.getTime() && vm.activity.finish.getTime() < vm.repeatDate.getTime()) {
+                        vm.activity.start.setDate(vm.activity.start.getDate() + 1);
+                        vm.activity.finish.setDate(vm.activity.finish.getDate() + 1);
+                        if (vm.repeatDays[vm.activity.start.getDay()]) {
+                            Activity.add(vm.activity).then(ok, fail);
+                            return;
+                        }
+                    }
+                }
```

Comment:
This is great and solves our problem!
There is a slight risk of putting the browser on a semi-infinite loop if someone tries to add an something valid until very far from now...